### ISSUE TITLE
Outputfile deletes one message

### ISSUE THREAD

**Issue by iemem15** [elastic/logstash#2806]
When using the output file plugin always deletes on message. Is there a way to fix it. I have been tested several times and does the same.

Thanks in advance

**Comment by jordansissel** [comment]
I think this is the same issue as https://github.com/logstash-plugins/logstash-output-file/issues/11

**Comment by iemem15** [comment]
Not at all.

This in the codec, not in the output, output works, codec doesn't

2015-05-18 0:24 GMT-05:00 Jordan Sissel <notifications@github.com>:

> Closed #2806 <https://github.com/elastic/logstash/issues/2806>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/logstash/issues/2806#event-307034053>.
>



-- 
Israel Elías Morales

**Comment by wiibaa** [comment]
@iemem15 file output is not respecting/using  codecs so the issue is really the flushing mechanism that will never write the last event as mentionned in https://github.com/logstash-plugins/logstash-output-file/blob/master/lib/logstash/outputs/file.rb#L179
````
 # every flush_interval seconds or so (triggered by events, but if there are no events there's no point flushing files anyway)
````
it should indeed be triggered by a timer not depending on events to avoid this case for "final" events.
